upgrade_test.py

```diff
@@ -1018,12 +1018,15 @@ def test_kubernetes_platform_upgrade(self):
             stress_cmd=self._cs_add_node_flag(self.params.get('stress_cmd_r'))))
 
         InfoEvent(message='Step4 - Upgrade kubernetes platform').publish()
-        upgrade_version = self.k8s_cluster.upgrade_kubernetes_platform()
+        upgrade_version, scylla_pool = self.k8s_cluster.upgrade_kubernetes_platform(
+            pod_objects=self.db_cluster.nodes,
+            use_additional_scylla_nodepool=True)
```

Comment:
Shouldn't we be doing this only of GKE ? and keep testing EKS in the same manner ?

I think that If there's two supported ways of doing upgrades, we should cover both.

Reply:
In case of the EKS we just didn't hit the same problem we had in GKE due to the different situation with disks.
So, need to start loading cluster enough to make a Scylla pod be migrated for more than an hour.

Then, on last operator meeting, I was told that the new approach is the only one we want to test.
So, I switched everything to the new approach, just kept the possibility to run the old one for case the decision changes.

Then, about the idea to cover both cases. I think, if we want to cover both cases, then need to do it per cluster.
And, since one of them is known to be broken on GKE. We should have following test coverage:
+ EKS using only existing node pools.
+ EKS using additional node pool.
+ GKE using additional node pool.

Reply:
> Then, on last operator meeting, I was told that the new approach is the only one we want to test.
> So, I switched everything to the new approach, just kept the possibility to run the old one for case the decision changes.

I wasn't there in that meeting, so if that's what agreed we are going to test. that's fine.
It should be clearly documented in operator, that that's the only supported way of upgrading k8s.

Reply:
The fact that the code supports both approaches already means it will be very easy to start testing second variant.
Even if it will be additional CI job, still small and easy changes.

Personally, I am leaning to having additional EKS job, so we will decide the need to test it only by either trigerring the job or not.